coalib/misc/Caching.py

```diff
@@ -0,0 +1,142 @@
+import os
+import calendar
+import time
+import appdirs
+import pickle
```

Comment:
Hm, I'm not completely sold on pickle, yet. It's nice that until now we have resided with human-readable files, although I get that pickle makes things a little easier in this case. And this is something that peolpe would most likely not interfere with besides deleting said file in case they suspect it leading to errors.